src/server/pmix_server_get.c

```diff
@@ -404,6 +413,29 @@ static pmix_status_t _satisfy_request(pmix_nspace_t *nptr, int rank,
      * having been committed */
     htptr = hts;
     PMIX_CONSTRUCT(&pbkt, pmix_buffer_t);
+
+    /* if they are asking about a rank from an nspace different
+     * from their own, then include a copy of the job-level info */
+    if (NULL != cd &&
+        0 != strncmp(nptr->nspace, cd->peer->info->nptr->nspace, PMIX_MAX_NSLEN)) {
+        cur_rank = PMIX_RANK_WILDCARD;
+        if (PMIX_SUCCESS != (rc = pmix_bfrop.pack(&pbkt, &cur_rank, 1, PMIX_INT))) {
+            PMIX_ERROR_LOG(rc);
+            PMIX_DESTRUCT(&pbkt);
+            cbfunc(rc, NULL, 0, cbdata, NULL, NULL);
+            return rc;
+        }
+        /* the client is expecting this to arrive as a byte object
+         * containing a buffer, so package it accordingly */
+        pbptr = &nptr->server->job_info;
+        if (PMIX_SUCCESS != (rc = pmix_bfrop.pack(&pbkt, &pbptr, 1, PMIX_BUFFER))) {
+            PMIX_ERROR_LOG(rc);
```

Comment:
What if later we will not find the rank information? we already have this buffer and we will need to just drop it.
I guess that for the performance reasons we want to pack this after we make sure that we have the response now and don't need to go for dmdx.